packages/flutter/lib/src/material/drawer.dart

```diff
@@ -217,7 +221,18 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   void _move(DragUpdateDetails details) {
-    final double delta = details.primaryDelta / _width;
+    if (_controller.isDismissed) {
+      final ModalRoute<dynamic> parentRoute = ModalRoute.of(context);
+      final bool canPop = parentRoute?.canPop ?? false;
+      if(canPop) {
+        Navigator.of(context).pop();
+        return;
+      }
+    }
```

Comment:
Can you explain this logic?  Why is this processing triggered on `_move`?  I would have expected it to be triggered based on an `addStatusListener` on the `_controller`.

I get that you're trying to keep the two drawers from interfering with each other.  I don't quite understand the interaction design you're going for.  Can you give a few example scenarios (i.e., sequences of user input) where something bad would happen without logic to prevent interference and then explain what you think should happen to resolve the interference and how your implementation achieves that.

As an example, it's quite possible to open both drawers without ever calling `_move`.  For example, just be calling `openDrawer` and `openEndDrawer` on the `ScaffoldState`.  Is that a case where we need to prevent interference?  Another input sequence to consider is when the user touches down on both drawer buttons and then releases them in sequence.  It should be the case that the first touch released should trigger that drawer to open and then I would expect the navigation caused by that drawer to cancel the second gesture, preventing the tap on the second button.

Reply:
The only intention behind having the processing triggered on _move is to have this logic (only one of the drawers open at a time) as a 'default' behavior.
There could be scenarios where customized implementations need to have both drawers opened at the same time.

Reply:
I feel like we don't have a solid interaction design for the two drawer case.  That's fine.  Maybe the right thing to do is leave it out of the first patch and add it later if/when it becomes obvious what we need to do.

Reply:
Okay. However, it would be great if we could have this behavior, since that is one of the requirements we've raised the PR for.